src/Microsoft.SqlTools.ServiceLayer/TableDesigner/TableDesignerService.cs

```diff
@@ -664,6 +666,27 @@ private bool GetBooleanValue(object value)
             return (bool)value;
         }
 
+        private bool GetIsNode(TableInfo tableInfo)
```

Comment:
I'd suggest just making this an extension method on  the table designer (and put in a separate static class file)

```
public static bool IsNode(this TableDesigner tableDesigner)
{
   return tableDesigner.TableViewModel.IsEdge
}
```

and then in GetPropertyBag just do

```
var tableDesigner = this.GetTableDesigner(tableInfo);
var metadata = new Dictionary<string, string>()
{
            {"IsEdge", tableDesigner.IsEdge },
            // other ones here
}
return metadata;
```

It would also be nice to have Metadata be an extension function on TableInfo, but I don't know if you can do that and be able to get the TableDesigner from that...

Reply:
We get the table designer class from dacfx, so can't change that class.  I can change the name to metadata.

Reply:
Extension methods are something you add in your own code, this isn't modifying the original class.

https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/extension-methods